packages/shared/src/components/feeds/FeedSettings/FeedSettingsCreate.tsx

```diff
@@ -0,0 +1,256 @@
+          (item) => item.message === labels.feed.error.feedLimit.api,
+        )
+      ) {
+        displayToast(labels.feed.error.feedLimit.client);
+
+        return;
+      }
+
+      if (
+        clientErrors.some(
+          (item) => item.message === labels.feed.error.feedNameInvalid.api,
+        )
+      ) {
+        displayToast(labels.feed.error.feedNameInvalid.api);
+
+        return;
+      }
+
+      displayToast(labels.error.generic);
+    },
+  });
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    completeAction(ActionType.CustomFeed);
+  }, [completeAction, user]);
+
+  const logStartEventRef = useRef(false);
+
+  useEffect(() => {
+    if (logStartEventRef.current) {
+      return;
+    }
+
+    logStartEventRef.current = true;
+
+    logEvent({
+      event_name: LogEvent.StartCustomFeed,
+    });
+  }, [logEvent]);
+
+  const onRequestClose = () => {
+    router.replace(webappUrl);
+  };
+
+  useEffect(() => {
+    if (!isPlus) {
+      router.replace(webappUrl);
+    }
+  }, [isPlus, router]);
+
+  if (!isPlus) {
+    return null;
+  }
+
+  return (
+    <Modal
+      isOpen
+      kind={Modal.Kind.FixedCenter}
+      size={Modal.Size.Small}
+      onRequestClose={onRequestClose}
+    >
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+
+          onSubmit(data);
+        }}
+      >
+        <ModalClose className="top-2" onClick={onRequestClose} type="button" />
+        <Modal.Header title="" showCloseButton={false}>
+          <CreateTitle className="hidden tablet:flex" />
+          <div className="flex w-full items-center justify-between gap-2 tablet:hidden">
+            <Button
+              type="button"
+              size={ButtonSize.Small}
+              variant={ButtonVariant.Tertiary}
+              onClick={onRequestClose}
+            >
+              Cancel
+            </Button>
+            <SubmitButton
+              loading={isSubmitPending || isAnimating}
+              disabled={!data.name}
+            />
+          </div>
+        </Modal.Header>
+        <Modal.Body className="flex flex-col gap-4">
+          <CreateTitle className="tablet:hidden" />
+          <TextField
+            className={{
+              container: 'w-full',
+            }}
+            name="name"
+            type="text"
+            inputId="feedName"
+            label="Give your feed a name"
+            required
+            maxLength={50}
+            valueChanged={(value) =>
+              setData((current) => ({ ...current, name: value }))
+            }
+          />
+          <div className="flex flex-col gap-4">
+            <Typography type={TypographyType.Body}>
+              <Typography className="inline-flex" bold>
+                Choose an icon
+              </Typography>{' '}
+              (optional)
+            </Typography>
+            <ul className="flex flex-wrap gap-4" role="radiogroup">
+              {emojiOptions.map((emoji) => (
```

Comment:
This whole thing is probably a copy of bookmark create right?

Reply:
yes 🙈